Jenkinsfile

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env groovy
+
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+
+/* These platforms correspond to labels in ci.jenkins.io, see:
+ *  https://github.com/jenkins-infra/documentation/blob/master/ci.adoc
+ */
+List platforms = ['linux', 'windows']
+Map branches = [:]
+
+for (int i = 0; i < platforms.size(); ++i) {
+    String label = platforms[i]
+    branches[label] = {
+        node(label) {
+            timestamps {
+                stage('Checkout') {
+                    checkout scm
+                }
+
+                stage('Build') {
+                    withEnv([
+                        "JAVA_HOME=${tool 'jdk8'}",
+                        "PATH+MVN=${tool 'mvn'}/bin",
+                        'PATH+JDK=$JAVA_HOME/bin',
+                    ]) {
+                        timeout(30) {
+                            sh 'mvn clean install -Dmaven.test.failure.ignore=true'
```

Comment:
Oh right, I forgot. It's still maddening that `bat` and `sh` are different steps